services/dbaas/users.go

```diff
@@ -182,12 +182,10 @@ func SetUserMongoData(d *schema.ResourceData, user *mongo.User) error {
 //nolint:golint
 func SetUserPgSqlData(d *schema.ResourceData, user *pgsql.UserResource) error {
 	resourceName := "PgSQL user"
-	d.SetId(*user.Id)
-	if user.Properties == nil {
-		return fmt.Errorf("expected properties in the response for the PgSql user with ID: %s, but received 'nil' instead", *user.Id)
-	}
-	if user.Properties.Username != nil {
-		if err := d.Set("username", *user.Properties.Username); err != nil {
+	d.SetId(user.Id)
+
+	if user.Properties.Username != "" {
```

Comment:
@cristiGuranIonos same here, I don't think we need this check.

Reply:
@adeatcu-ionos not sure if this was the case when I left the comment and I didn't check properly, but we need this check here because `Name` is a pointer.